pkg/plugin/manager.go

```diff
@@ -0,0 +1,196 @@
+package plugin
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// The plugin manager retrieves the plugins available for the current cluster
+// by navigating into the filesystem. It requires a `plugin.yaml` file for
+// each plugin, and can create this file if it doesn't exist yet.
+type Manager struct {
+	Fs     afero.Fs
+	Logger *logrus.Logger
+	Dir    string
+}
+
+// Structure representing information for old plugins found in Universe packages.
+type Package struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Version     string `json:"version"`
+}
+
+// Plugins returns the plugins associated with the current cluster.
+func (m *Manager) Plugins() []*Plugin {
+	plugins := []*Plugin{}
+
+	pluginsDirHandle, err := m.Fs.Open(m.Dir)
+	if err != nil {
+		// If we fail to open the directory, we return an empty list.
+		return plugins
+	}
+	defer pluginsDirHandle.Close()
+
+	pluginDirs, err := pluginsDirHandle.Readdir(0)
+	if err != nil {
+		// If we fail to open the directory, we return an empty list.
+		return plugins
+	}
+
+	for _, pluginDir := range pluginDirs {
+		if pluginDir.IsDir() {
+			plugin := &Plugin{}
+
+			// Set plugin directories.
+			plugin.dir = filepath.Join(m.Dir, pluginDir.Name())
+
+			if runtime.GOOS == "windows" {
+				plugin.binDir = filepath.Join(plugin.dir, "env", "Scripts")
```

Comment:
Should we support the `bin` dir on Windows too?

Reply:
We have tested the current DC/OS CLI on Windows and the standard path for subcommands replace `bin` by `Scripts`.